controllers/capabilities/backend_threescale_reconciler.go

```diff
@@ -447,18 +450,32 @@ func (t *BackendThreescaleReconciler) reconcileMatchedMetrics(matchedMap map[str
 	return nil
 }
 
-func (t *BackendThreescaleReconciler) syncMappingRules(_ interface{}) error {
+func (t *BackendThreescaleReconciler) syncMappingRules(_ interface{}) (error, []string) {
+	var warnings []string
 	desiredKeys := make([]string, 0, len(t.backendResource.Spec.MappingRules))
 	desiredMap := map[string]capabilitiesv1beta1.MappingRuleSpec{}
 	for _, spec := range t.backendResource.Spec.MappingRules {
-		key := fmt.Sprintf("%s:%s", spec.HTTPMethod, spec.Pattern)
+		key := fmt.Sprintf("%s:%s:%s", spec.HTTPMethod, spec.Pattern, spec.MetricMethodRef)
```

Comment:
the difference with my approach (validation rules in Product) that you adding just first rule (from duplicates), and me - last. But main issue not this, but UI, and need decide how to handle it. We can remove duplication from Existing rules (that including alsso adding from UI), but seems need add some env var to allow user to decide about cleanup. It will allow control UI without changes in Porta (UI) and porta-go-clint. I think at first step - need just address the BUG, and next - extend the process of validation and control.